airflow/operators/file_transfer.py

```diff
@@ -0,0 +1,76 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+import shutil
+from typing import TYPE_CHECKING
+
+from airflow.io.fsspec import FsspecFileIO
+from airflow.models import BaseOperator
+
+if TYPE_CHECKING:
+    from airflow.io import InputFile, OutputFile
+    from airflow.utils.context import Context
+
+
+class FileTransfer(BaseOperator):
```

Comment:
Should this be named e.g. `TransferOperator` to match existing name style?

Reply:
We are not as consistent as you might think ;-). `GenericTransfer` does the same now. Given that `GenericTransfer` (which isn't really generic btw) becomes very close to `TransferOperator`, I'd prefer `FileTransferOperator` - which might be overly verbose or keep as is. `Transfer` does include the notion of `Operator` I think, but ymmv.

Reply:
I didn’t even know we have GenericTransfer. FWIW I’m thinking about potentially extending this operator to cover copying from SQL to a (say Parquet) file, but that would make the “File” part a bit awkward. But maybe that lives better in a separate operator instead. Not sure. For the purpose this covers at the momemnt, FileTransferOperator sounds good to me.

Reply:
As described in the PR header - at the moment scope is file IO only, but the same pattern can be extended (in future) to build a TableToFile and FileToTableOperator on top using the same IO and for example Parquet in between. So for me this is looking like a goos start.

Reply:
Will rename to `FileTransferOperator`. 

I like the extension idea, but would like to put the DataFrame API in between so that you would do: 

SQL -> Stream -> DataFrame -> Write (iceberg, parquet, csv) -> ObjectStorage 

or opposite

ObjectStorage -> Read (iceberg, parquet, csv) -> DataFrame -> Stream -> SQL

This would anything that creates or understands a DataFrame Stream to write out or read a structured format. This PR should create the initial ground work for that, but the high level APIs are out of scope.